ksql-execution/src/main/java/io/confluent/ksql/execution/function/udaf/KudafInitializer.java

```diff
@@ -13,23 +13,27 @@
  * specific language governing permissions and limitations under the License.
  */
 
-package io.confluent.ksql.function.udaf;
+package io.confluent.ksql.execution.function.udaf;
 
+import com.google.common.collect.ImmutableList;
 import io.confluent.ksql.GenericRow;
-import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 import org.apache.kafka.streams.kstream.Initializer;
 
 public class KudafInitializer implements Initializer<GenericRow> {
 
-  private final List<Supplier> aggValueSuppliers = new ArrayList<>();
+  private final List<Supplier> aggValueSuppliers;
   private final int nonAggValSize;
 
-  public KudafInitializer(final int nonAggValSize) {
+  public KudafInitializer(final int nonAggValSize, final List<Supplier> aggValueSuppliers) {
```

Comment:
nit: can we call this param and the field by what it actually is... which is the `initialValuesSupplier`, right?